### ISSUE TITLE
Document input in `score_genes()`/`score_genes_cell_cycle()`

### ISSUE THREAD

**Issue by lazappi** [scverse/scanpy#2909]
### What kind of feature would you like to request?

Other?

### Please describe your wishes

Currently, `score_genes()`/`score_genes_cell_cycle()` do not have any examples and specifically no documentation of what the input transformation of the data should be, just a link to a (very old) notebook.

The notebook says "Log-transformation of data and scaling should always be performed before scoring" but I suspect many users (like me) have missed this (especially the scaling step!). It would be great if this was stated in the function documentation with an example.

**Comment by lazappi** [comment]
From a quick look at the Seurat code this was borrowed from, I think normalised (but not scaled) data is used, so maybe the scaling isn't needed https://github.com/satijalab/seurat/blob/656fc8b562d53e5d0cedda9e09d9dda81e8c00e9/R/utilities.R#L192. Either way it would be good for this to be clear in the documentation.

**Comment by jonesr18** [comment]
With the way scaling works (scale gene expression per mean and variance across observations), isn't it weird to then do the scores (which are calculated per-cell)? This essentially gives you a score of how each gene is expressed compared to other cells, which may not preserve the rankings of gene expression within a given cell. I see the benefit though of normalizing the mean+variance of each gene to limit outsized impacts from genes that are relatively high across the board.

**Comment by flying-sheep** [comment]
Good point about the old notebook @lazappi, time to bring it into 2025: https://github.com/scverse/scanpy-tutorials/pull/195

Looks like `scale` isn’t necessary to have equivalent results:

<img width="1415" height="1248" alt="Image" src="https://github.com/user-attachments/assets/933b0205-42d7-475e-bd8f-718e4adc5e44" />
